### ISSUE TITLE
Bioc rworkflows fail

### ISSUE THREAD

**Issue by TuomasBorman** [neurogenomics/rworkflows#130]
Hello!

We are using rworkflows in our Bioconductor packages. We noticed maybe week ago that Ubuntu (and MAC) workflow started to fail; we did not do any changes. We thought that the problem might resolve itself, but it has not.

Here is our workflow: https://github.com/microbiome/mia/blob/devel/.github/workflows/rworkflows.yml

Here is the result of the workflow run: https://github.com/microbiome/mia/actions/runs/8935442827/job/24543956269

I have not been able to fix the problem, but it seems that:

1. The workflow fails due to WARNINGS
2. WARNINGS are caused by warning messages "package X was built under R version 4.5.0"

Bioconductor development should be done against R-release (R-4.4.0) currently. So it seems that rworkflows picks the wrong R version which causes this problem (some packages are built in R-4.4.0 and some in R-4.5.0).

Thanks in advance!

BR,
Tuomas Borman

**Comment by bschilder** [comment]
The timing suggests this may be due to the recent release of Bioc 3.19
https://bioconductor.org/news/bioc_3_19_release/

There's always a lag between the Bioc release cycle and the Bio Docker container update:
https://github.com/neurogenomics/rworkflows?tab=readme-ov-file#bioconductor_docker

**Comment by bschilder** [comment]
Bioc version picked by rworkflows are pulled directly from Bioc:

```R
rworkflows::bioc_r_versions()
```
```
$devel
$devel$bioc
[1] "3.20"

$devel$r
[1] "4.4.0"


$release
$release$bioc
[1] "3.19"

$release$r
[1] "4.4.0"


$r_ver_for_bioc_ver
$r_ver_for_bioc_ver$`1.6`
[1] "2.1"

$r_ver_for_bioc_ver$`1.7`
[1] "2.2"

$r_ver_for_bioc_ver$`1.8`
[1] "2.3"

$r_ver_for_bioc_ver$`1.9`
[1] "2.4"

$r_ver_for_bioc_ver$`2.0`
[1] "2.5"

$r_ver_for_bioc_ver$`2.1`
[1] "2.6"

$r_ver_for_bioc_ver$`2.2`
[1] "2.7"

$r_ver_for_bioc_ver$`2.3`
[1] "2.8"

$r_ver_for_bioc_ver$`2.4`
[1] "2.9"

$r_ver_for_bioc_ver$`2.5`
[1] "2.10"

$r_ver_for_bioc_ver$`2.6`
[1] "2.11"

$r_ver_for_bioc_ver$`2.7`
[1] "2.12"

$r_ver_for_bioc_ver$`2.8`
[1] "2.13"

$r_ver_for_bioc_ver$`2.9`
[1] "2.14"

$r_ver_for_bioc_ver$`2.10`
[1] "2.15"

$r_ver_for_bioc_ver$`2.11`
[1] "2.15"

$r_ver_for_bioc_ver$`2.12`
[1] "3.0"

$r_ver_for_bioc_ver$`2.13`
[1] "3.0"

$r_ver_for_bioc_ver$`2.14`
[1] "3.1"

$r_ver_for_bioc_ver$`3.0`
[1] "3.1"

$r_ver_for_bioc_ver$`3.1`
[1] "3.2"

$r_ver_for_bioc_ver$`3.2`
[1] "3.2"

$r_ver_for_bioc_ver$`3.3`
[1] "3.3"

$r_ver_for_bioc_ver$`3.4`
[1] "3.3"

$r_ver_for_bioc_ver$`3.5`
[1] "3.4"

$r_ver_for_bioc_ver$`3.6`
[1] "3.4"

$r_ver_for_bioc_ver$`3.7`
[1] "3.5"

$r_ver_for_bioc_ver$`3.8`
[1] "3.5"

$r_ver_for_bioc_ver$`3.9`
[1] "3.6"

$r_ver_for_bioc_ver$`3.10`
[1] "3.6"

$r_ver_for_bioc_ver$`3.11`
[1] "4.0"

$r_ver_for_bioc_ver$`3.12`
[1] "4.0"

$r_ver_for_bioc_ver$`3.13`
[1] "4.1"

$r_ver_for_bioc_ver$`3.14`
[1] "4.1"

$r_ver_for_bioc_ver$`3.15`
[1] "4.2"

$r_ver_for_bioc_ver$`3.16`
[1] "4.2"

$r_ver_for_bioc_ver$`3.17`
[1] "4.3"

$r_ver_for_bioc_ver$`3.18`
[1] "4.3"

$r_ver_for_bioc_ver$`3.19`
[1] "4.4"

$r_ver_for_bioc_ver$`3.20`
[1] "4.4"
```

**Comment by bschilder** [comment]
> Alright, thanks. That sounds exactly how it should be. However, the error messages suggested mismatch between R versions. We check if we can fix our workflow issue

I agree, the specific message you reported does still have me a bit puzzled. 
One thing that might help is removing the ubuntu-latest rspm line, or replacing it with `rspm: ~`. This way, it will use the default rspm included within the Bioc Docker container.
https://github.com/microbiome/mia/blob/7a0b0ead9d82087f86e29e4867b3412f88303cd7/.github/workflows/rworkflows.yml#L30

**Comment by TuomasBorman** [comment]
We are trying to troubleshoot the problem in this PR https://github.com/microbiome/mia/pull/544

The documentation of rworkflows indicates that secrets are needed for some options such as Docker (we have not enabled it yet). We have been using rworkflows for a while, and it has worked before. Has something changed because it seems that github-actions does not have access anymore to deploy documentation (we have not had secrets before either)?

```
remote: Permission to microbiome/mia.git denied to github-actions[bot].
fatal: unable to access 'https://github.com/microbiome/mia.git/': The requested URL returned error: 403
Running post deployment cleanup jobs 
```

**Comment by TuomasBorman** [comment]
We were able to solve the issue. Something related to cache and this R/Bioc cycle. That permission thing resolved when the build was run in devel branch.
